src/parser/creative_companion_parser.js

```diff
@@ -14,79 +14,89 @@ import { parserUtils } from './parser_utils';
  */
 export function parseCreativeCompanion(creativeElement, creativeAttributes) {
   const creative = new CreativeCompanion(creativeAttributes);
+  creative.required = creativeElement.getAttribute('required') || null;
 
-  parserUtils
+  creative.variations = parserUtils
     .childrenByName(creativeElement, 'Companion')
-    .forEach(companionResource => {
-      const companionAd = new CompanionAd();
-      companionAd.id = companionResource.getAttribute('id') || null;
-      companionAd.width = companionResource.getAttribute('width');
-      companionAd.height = companionResource.getAttribute('height');
-      companionAd.companionClickTrackingURLTemplates = [];
+    .map(companionResource => {
+      const attributes = {};
+      const companionAttrs = companionResource.attributes;
+      for (let i = 0; i < companionAttrs.length; i++) {
+        attributes[companionAttrs[i].nodeName] = companionAttrs[i].nodeValue;
+      }
+      const companionAd = new CompanionAd(attributes);
 
-      parserUtils
-        .childrenByName(companionResource, 'HTMLResource')
-        .forEach(htmlElement => {
-          companionAd.type =
-            htmlElement.getAttribute('creativeType') || 'text/html';
-          companionAd.htmlResource = parserUtils.parseNodeText(htmlElement);
-        });
+      companionAd.htmlResource =
+        parserUtils.parseNodeText(
+          parserUtils.childByName(companionResource, 'HTMLResource')
+        ) || null;
```

Comment:
According to the VAST spec 
> Each <Companion> element must specify at least one resource file that may be one of: 
> `<StaticResource>`, `<IFrameResource>` or `<HTMLResource>`

What is the approach we want to take when multiple resources are provided?

Reply:
I didn't understand from the specs whether you could have multiple `StaticResources` for example, or just multiple resources of different types. either way we were still only assigning one resource per type previously

Reply:
added support for multiple of the same resources